File: modified_scripts/scala/matrix_multiplication/deepseek-coder_6.7b.scala

def scalarMultiply(x: Array[Array[Double]], y: Array[Array[Double]], row1: Int, col1: Int, row2: Int, col2: Int): Double = {
  var sum = 0.0
  for (i <- 0 until row1; j <- 0 until col1) yield {
    sum += x(i)(j) * y(i)(j)
  }
  sum
}

def matrixMultiplication(a: Array[Array[Double]], b: Array[Array[Double]], r1: Int, c1: Int, r2: Int, c2: Int): Array[Array[Double]] = {
  val res = new Array[Array[Double]](r1)
  for (i <- res.indices) yield {
    res(i) = new Array[Double](c2)
  }

  if (c1 != r2) throw new Exception("Column of first matrix is not equal to row of second.")
  
  for (i <- res.indices; j <- res(0).indices) yield {
    res(i)(j) = scalarMultiply(a, b, r1, c1, r2, c2)
  }
  res
}

File: modified_scripts/julia/matrix_multiplication/deepseek-coder_6.7b.jl

function matrix_multiplication(A::AbstractMatrix{T}, B::AbstractMatrix{T}) where {T <: Number}
    m, p = size(A)
    n, q = size(B)
    
    @assert p == n "First matrix column dimension must equal second matrix row dimension"

    C = similar(A, T, m, q)
    for i = 1:m
        for j = 1:q
            C[i,j] = sum(A[i,k]*B[k,j] for k = 1:p)
        end
    end
    
    return C
end
